### PR TITLE
Fix/wallet connect transaction fixes

### PR DESCRIPTION
**Pull request by poocart**
This PR includes:
- Added check if it's enough ether when confirming and display warning if false
- Added gas price and gas limit to all ERC20 transactions
- Added check if Wallet Connect ERC20 transaction request token is supported by Pillar Wallet and display warning if false
- Fixed incoming Wallet Connect transaction amount for ETH transactions
- Added default gas limit for Wallet Connect incoming request payload
- Fixed Wallet Connect ERC20 transaction approval and send

### CODE REVIEW

File: src/screens/WalletConnect/WalletConnectCallRequest.js [hunk 1]
```diff
@@ -95,10 +109,18 @@ class WalletConnectCallRequestScreen extends React.Component<Props, State> {
       if (matchingAssets && matchingAssets.length) {
         asset = matchingAssets[0]; // eslint-disable-line
         symbol = asset.symbol; // eslint-disable-line
-        amount = utils
-          .bigNumberify(data.substring(73))
-          .mul(utils.bigNumberify('10').pow(asset.decimals));
+        const iface = new Interface(ERC20_CONTRACT_ABI);
```
Comment (jsidorenko) [inline src/screens/WalletConnect/WalletConnectCallRequest.js]: Pls check how all this work with tokens those have a non-standard amount of decimals (not 18)

File: src/screens/WalletConnect/WalletConnectCallRequest.js [hunk 2]
```diff
@@ -316,13 +359,14 @@ class WalletConnectCallRequestScreen extends React.Component<Props, State> {
...
 const mapStateToProps = ({
-  assets: { supportedAssets },
+  assets: { supportedAssets, balances },
   contacts: { data: contacts },
   session: { data: session },
 }) => ({
   contacts,
   session,
   supportedAssets,
+  balances,
```
Comment (jsidorenko) [inline src/screens/WalletConnect/WalletConnectCallRequest.js]: When we merge this into smart-wallet branch, we need to use a selector to get the balance

Comment (poocart) [reply src/screens/WalletConnect/WalletConnectCallRequest.js]: 👍

File: src/screens/WalletConnect/WalletConnectCallRequest.js (hunk 1 above)
Comment (grilix) [reply src/screens/WalletConnect/WalletConnectCallRequest.js]: That's a good catch. Do we have that information on the assets list?

Comment (grilix) [reply src/screens/WalletConnect/WalletConnectCallRequest.js]: Actually, the contract doesn't seem to include the decimals and then we're doing `amount = parseFloat(utils.formatUnits(methodValue, asset.decimals));`. Is there any example you can come up with where we could have a problem regarding the decimal points?

Comment (jsidorenko) [reply src/screens/WalletConnect/WalletConnectCallRequest.js]: Everything should be fine here, I just missed that part during review: `amount = parseFloat(utils.formatUnits(methodValue, asset.decimals));`

File: src/services/assets.js [hunk 3]
```diff
@@ -62,21 +64,31 @@ function contractHasMethod(contractCode, encodedMethodName) {
...
+    gasLimit,
+    gasPrice,
   } = options;
   wallet.provider = getEthereumProvider(NETWORK_PROVIDER);
   const contract = new Contract(contractAddress, ERC20_CONTRACT_ABI, wallet);
-  if (decimals > 0) {
-    return contract.transfer(to, utils.parseUnits(amount.toString(), decimals), { nonce });
-  }
-  return contract.transfer(to, utils.bigNumberify(amount.toString()), { nonce });
+  const decimals = defaultDecimals > 0
```
Comment (alazarevski) [inline src/services/assets.js]: for naming purposes, this should be parsedAmount right ?

File: src/screens/WalletConnect/WalletConnectCallRequest.js (hunk 1 above)
Comment (poocart) [reply src/screens/WalletConnect/WalletConnectCallRequest.js]: Changed this to string as last decimals are being rounded when converted to JavaScript number.

File: src/services/assets.js (hunk 3 above)
Comment (poocart) [reply src/services/assets.js]: Thanks, changed it to `contractAmount`.

**Pull request merged**